tests/test_cfg/test_z3_constraints.py

```diff
@@ -18,8 +18,12 @@ def func(x: int, y: int) -> int:
     cfg = _create_cfg(src, "func")
     x = z3.Int("x")
     y = z3.Int("y")
-    expected = {0: [z3.And(x > 0, y > 0), x >= y]}
-    assert all(edge.z3_constraints == expected for edge in cfg.get_edges())
+    expected = z3.simplify(z3.And(x > 0, y > 0, x >= y))
+    assert all(
+        edge.z3_constraints[0][0] == expected
+        for edge in cfg.get_edges()
+        if 0 in edge.z3_constraints and edge.z3_constraints[0]
```

Comment:
Adding extra guards to these tests shouldn't be necessary. The code change did update the value of `edge.z3_constraints` but should not change the overall structure of the list, nor the number of edges.

In general you should avoid unnecessary like this, as they either add confusion about what the expected behaviour is, or (in the case of failing tests) can mask deeper issues with the code.

The same comment applies throughout; I would expect to see only changes to the expected value of the constraints by calling `z3.simplify`, but no other test changes.